edge-modules/ModuleRestarter/src/Program.cs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace ModuleRestarter
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    class Program
+    {
+        static readonly ILogger Log = Logger.Factory.CreateLogger<Program>();
+        static readonly string ServiceClientConnectionString = Settings.Current.ServiceClientConnectionString;
+        static readonly string DeviceId = Settings.Current.DeviceId;
+        static readonly string DesiredModulesToRestartCSV = Settings.Current.DesiredModulesToRestartCSV;
+        static readonly int RandomRestartIntervalInMins = Settings.Current.RandomRestartIntervalInMins;
+
+        public static int Main() => MainAsync().Result;
+
+        static async Task<int> MainAsync()
+        {
+            Log.LogInformation($"Starting module restarter with the following settings:\r\n{Settings.Current}");
+
+            if (string.IsNullOrEmpty(DesiredModulesToRestartCSV))
+            {
+                Log.LogInformation("No modules specified to restart. Stopping.");
+                return 0;

```

Comment:
if the program exits, I believe the docker container will exit as well. Will this be forced to restart by edgeAgent as restart policy is set to always?

Reply:
You need to think about what behavior you expected.  For example, you can still run and check every X period if the environment variable may be updated.

Reply:
I did not know the container will stop. EdgeAgent will definitely restart so this absolutely needs to change.

Reply:
I don't think it's a good idea to wait for new environment variables. Wouldn't the only way this could happen is from another deployment? In that case, the module is restarting anyway. I feel like the cleaner solution is to use Preconditions.Check() to throw an exception in the case of bad args, which will not restart the module.